pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -175,46 +189,57 @@ func (c StaticResourceController) Sync(ctx context.Context, syncContext factory.
 		return nil
 	}
 
-	errors := []error{}
+	var errors, notFoundErrors []error
 	directResourceResults := resourceapply.ApplyDirectly(c.clients, syncContext.Recorder(), c.manifests, c.files...)
 	for _, currResult := range directResourceResults {
+		if c.ignoreNotFound && apierrors.IsNotFound(currResult.Error) {
+			notFoundErrors = append(notFoundErrors, fmt.Errorf("%q (%T): %v", currResult.File, currResult.Type, currResult.Error))
+			continue
+		}
 		if currResult.Error != nil {
 			errors = append(errors, fmt.Errorf("%q (%T): %v", currResult.File, currResult.Type, currResult.Error))
 			continue
 		}
 	}
 
-	if len(errors) > 0 {
-		message := ""
-		for _, err := range errors {
-			message = message + err.Error() + "\n"
+	if c.ignoreNotFound {
+		// If ignored, save NotFound errors in <controller name>Error condition, so the whole operator is not Degraded.
+		err := c.SetFailureCondition(fmt.Sprintf("%sError", c.name), notFoundErrors)
```

Comment:
Please continue to use the *Degraded condition, but set the message while saying Degraded=false

Reply:
Done